apps/ledger-live-mobile/src/components/DeviceAction/InstallSetOfApps/index.tsx

```diff
@@ -38,14 +44,36 @@ const InstallSetOfApps = ({
 }: Props & { remountMe: () => void }) => {
   const [userConfirmed, setUserConfirmed] = useState(false);
   const productName = getDeviceModel(selectedDevice.modelId).productName;
+  const lastSeenDevice: DeviceModelInfo | null | undefined = useSelector(
+    lastSeenDeviceSelector,
+  );
+  const lastDeviceProductName = lastSeenDevice?.modelId
+    ? getDeviceModel(lastSeenDevice?.modelId).productName
+    : lastSeenDevice?.modelId;
+
+  const shouldRestoreApps = restore && !!lastSeenDevice;
+
+  const getLastSeenDeviceAppNames = useCallback(() => {
+    if (!lastSeenDevice) {
+      return null;
+    }
+    return lastSeenDevice.apps.map(app => app.name);
+  }, [lastSeenDevice]);
+
+  const getDependencies = useCallback(() => {
+    if (shouldRestoreApps) {
+      return getLastSeenDeviceAppNames() || [];
```

Comment:
a callback or a memo ? the function `getLastSeenDeviceAppNames` is only used below in `getDependencies`

To simplify a bit, should `getDependencies` be a memoized array (with `useMemo`) named for ex `dependenciesToInstall`, that returns `lastSeenDevice.apps.map(app => app.name)` or `[]` if `shouldRestoreApps` is true, and `dependencies` otherwise ?

Reply:
WDYT ?

Reply:
Done too